src/asset-proxy.ts

```diff
@@ -0,0 +1,32 @@
+import { Url, getPlaceholder, isImageResponse } from './utils'
+
+export async function assetProxy(request: Request): Promise<Response | null> {
+  const url = Url.fromRequest(request)
+
+  if (url.subdomain !== 'asset-proxy') return null
+  if (url.pathname !== '/src') return null
+
+  const urlParam = url.searchParams.get('url')
+
+  if (!urlParam) return getPlaceholder()
+
+  let assetUrl: Url
+
+  try {
+    assetUrl = new Url(urlParam)
```

Comment:
if we stick with encoded we need a `decodeURIComponent` here.

Reply:
I'm not sure I understand your suggestion...

If you mean
```typescript
new Url(decodeURIComponent(urlParam))
```
it is unnecessary because the ` const url = Url.fromRequest(request)`  of line 4 has already done that (don't ask me why): 

```typescript
// __tests__/asset-proxy.ts
const encodedUrl = encodeURIComponent('https://whatever.org/image')
console.log({encodedUrl}}
  const response = await env.fetch({
    subdomain: 'asset-proxy',
    pathname: `/src?url=${encodedUrl}`,
  })
//   console.log
//    { encodedUrl: 'https%3A%2F%2Fwhatever.org%2Fimage' }


//
  const url = Url.fromRequest(request)
// ...
  const urlParam = url.searchParams.get('url')
  console.log({urlParam})
//  console.log
//    { urlParam: 'https://whatever.org/image' }
```


But I think you mean 
```
  const originalResponse = await fetch(
    encodeURIComponent(assetUrl.toString()),
    {
      cf: { cacheTtl: 24 * 60 * 60 },
    },
  )
```
Isn' that? Already, it is done in ffb79ba

Reply:
Sorry, I had to revert https://github.com/serlo/cloudflare-worker/commit/ffb79ba10c92c77ec5fcbf4728a9dc841391104d. Interesting enough it was throwing an error at the fetch call

Reply:
That's actually what I meant, I thought we have to manually decode here. 
So now I understand it better and am happy with the solution, thank you :)